lens/dict.rkt

```diff
@@ -0,0 +1,16 @@
+#lang racket/base
+
+(provide dict-ref-lens)
+
+(require racket/dict fancy-app "base/main.rkt")
+(module+ test
+  (require rackunit))
+
+(define (dict-ref-lens key)
+  (make-lens (dict-ref _ key)
+             (dict-set _ key _)))
```

Comment:
This seems like a common pattern. A ref and set function that take the target, location, and new view in that order. Maybe a helper like `make-ref-set-lens` would be useful? (Or maybe the appropriate fix is for everyone to just use [generic collections](https://github.com/lexi-lambda/racket-alexis-collections)